frontend/public/components/secrets/create-secret.tsx

```diff
@@ -59,116 +84,258 @@ class BaseEditSecret_ extends SafetyFirst<BaseEditSecretProps_, BaseEditSecretSt
     const { kind, metadata } = this.state.secret;
     this.setState({ inProgress: true });
 
-    const newSecret = _.assign({}, this.state.secret, {stringData: this.state.stringData});
+    const newSecret = _.assign({}, this.state.secret, {stringData: this.state.stringData}, {type: this.state.type});
     const ko = kindObj(kind);
     (this.props.isCreate
       ? k8sCreate(ko, newSecret)
       : k8sUpdate(ko, newSecret, metadata.namespace, newSecret.metadata.name)
     ).then(() => {
       this.setState({inProgress: false});
       history.push(formatNamespacedRouteForResource('secrets'));
-    },
-    err => this.setState({error: err.message, inProgress: false})
-    );
+    }, err => this.setState({error: err.message, inProgress: false}));
   }
   render () {
-    const title = `${this.props.titleVerb} ${_.upperFirst(this.state.secretType)} Secret`;
-    const { saveButtonText } = this.props;
-
-    const explanation = 'Webhook secrets allow you to authenticate a webhook trigger.';
-    const subform = <WebHookSecretSubform onChange={this.onDataChanged.bind(this)} stringData={this.state.stringData} />;
-
+    const title = `${this.props.titleVerb} ${_.upperFirst(this.state.secretTypeAbstraction)} Secret`;
     return <div className="co-m-pane__body">
       <Helmet>
         <title>{title}</title>
       </Helmet>
-      <form className="co-m-pane__body-group" onSubmit={this.save}>
+      <form className="co-m-pane__body-group create-secret-form" onSubmit={this.save}>
```

Comment:
Prefer to prefix class names to avoid collisions with dependencies and for consistency with existing styles: `co-create-secret-form`